src/gui/guiScrollBar.cpp

```diff
@@ -183,36 +183,41 @@ void GUIScrollBar::draw()
 			slider_rect.LowerRightCorner.Y =
 					slider_rect.UpperLeftCorner.Y + thumb_size;
 		}
-		skin->draw3DButtonPaneStandard(this, slider_rect, &AbsoluteClippingRect);
+		
+		// Clip scrollbar so it doesn't show behind buttons for elastic scrolling
+		if (up_button->isVisible() || down_button->isVisible()) {
+            core::rect<s32> clip = AbsoluteClippingRect;
+            if (up_button->isVisible())
+                clip.UpperLeftCorner.Y = up_button->getAbsolutePosition().LowerRightCorner.Y;
+            if (down_button->isVisible())
+                clip.LowerRightCorner.Y = down_button->getAbsolutePosition().UpperLeftCorner.Y;
+            skin->draw3DButtonPaneStandard(this, slider_rect, &clip);
+        } else {
+            // Just clip it directly
+            skin->draw3DButtonPaneStandard(this, slider_rect, &AbsoluteClippingRect);
+        }
 	}
 	IGUIElement::draw();
 }
 
-static inline s32 interpolate_scroll(s32 from, s32 to, f32 amount)
+f32 GUIScrollBar::interpolate_scroll(f32 from, f32 to, f32 amount)
 {
-	s32 step = core::round32((to - from) * core::clamp(amount, 0.001f, 1.0f));
-	if (step == 0)
-		return to;
+	f32 step = (to - from) * (amount * (last_delta_ms / 16.667f));
```

Comment:
FPS comment needs to be here. (explains magic num)

Also, why not round? setPosRaw doesn't take floats.

Reply:
Setting the position and actually calculating the interpolated position I believe something I did intentionally. If you round then it's going to be difficult to actually use setPosRaw. setPosRaw is the displayed position, iirc from working on this PR.